Wallets

Dm1triiSmirnov opened the pull request:
In this pull request:
1. Hide secret information from settings. py (secret_key, login and password for DB)
2. Add features for wallet: to create wallets, get list of all wallets and retrieve/delete single wallet. Also modifiied wallet/serializers.py: add limits to create not more then 5 wallets and add bonus when user create new wallet

Dm1triiSmirnov committed:
add features to create wallets, get list of all wallets and retrieve/delete single wallet

Dm1triiSmirnov committed:
modifiy wallet/serializers.py, add limits to create not more then 5 wallets and bonus for new wallets

VladKli commented on wallet/serializers.py:
```diff
@@ -16,3 +16,23 @@ class Meta:
+            if RUB - balance=100.00
+        """
+
+        print(validated_data)
+        if Wallet.objects.filter(user__exact=validated_data.get("user")).count() >= 5:
+            raise Exception("User can't create more than 5 wallets")
+        if validated_data["currency"] in ["USD", "EUR"]:
+            validated_data["balance"] += 3
+        elif validated_data["currency"] == "RUB":
+            validated_data["balance"] += 100
```
it's better to create constants in the wallet model for the bank bonuses and max amount of wallets.
something like this 
```
MAX_WALLETS = 5
BANK_BONUS_USD_EUR = 3.00
BANK_BONUS_RUB = 100.00
```
then you will be able to get these values as an attribute of the wallet model.

remove print, please

Dm1triiSmirnov replied:
Done

Dm1triiSmirnov committed:
add constants for max wallets and bonuses

Dm1triiSmirnov committed:
add editable=False to the name field in wallet/models.py

Dm1triiSmirnov committed:
Merge branch 'master' into wallets

# Conflicts:
#	drf_project/settings.py

Dm1triiSmirnov committed:
modify wallet:
1. Allow to show list of the wallets only for current user
2. Hide user field during creating of new wallet

VladKli commented on wallet/models.py:
```diff
@@ -5,10 +5,13 @@
+BANK_BONUS_RUB = 100
 
 
 class Wallet(models.Model):
```
please add` __str__` method to the class

Dm1triiSmirnov replied:
Done

VladKli commented on wallet/models.py:
```diff
@@ -21,8 +24,8 @@ def generate_wallet_name():
-    )  # noqa E501
+        max_length=8, unique=True, default=generate_wallet_name, editable=False
+    )
     type = models.CharField(choices=WALLET_TYPE, max_length=100)
     currency = models.CharField(choices=CURRENCY, max_length=3)
```
please, set default values for `type` and `currency` fields
e.g. type=visa, currency=euro

Dm1triiSmirnov replied:
Done

VladKli commented on wallet/views.py:
It's better to add `permission_classes = [permissions.IsAuthenticated]` for the views.
So only logged in user can create wallets. 

In the WalletsRetrieveDestroyViewSet class we need to check if it's a user's wallet. because now the anonymous user can delete and get any wallet.

Dm1triiSmirnov replied:
Done. Also modified retrive method to show for user only his own wallet details

Dm1triiSmirnov committed:
fix wallet/models.py: add __str__ method; add default fields

Dm1triiSmirnov committed:
modify wallet/view.py: fix permissions for wallet; rewrite retrive & destroy methods

VladKli commented on wallet/serializers.py:
```diff
@@ -16,3 +20,30 @@ class Meta:
+            if RUB - balance=100.00
+        """
+
+        if (
+            Wallet.objects.filter(
+                user__exact=validated_data.get("user")
+            ).count()  # noqa E501
+            >= MAX_WALLETS
+        ):
+            raise Exception("User can't create more than 5 wallets")
```
please, create a separate function for this.

Dm1triiSmirnov committed:
fix wallets. Move check for wallets limit to views
